File: 프로그래머스/lv1/12943. 콜라츠 추측/콜라츠 추측.js
function solution(num) {
    let result = 0;
    
    while (num !== 1) {
        if (!(num % 2)) {
            num /= 2;
        } else {
            num = num * 3 + 1;
        }
        
        result++;
        
        if (result > 500) return -1;
    }
    
    return result;
}

File: 프로그래머스/1/12943. 콜라츠 추측/콜라츠 추측.java
class Solution {
    public int solution(int num) {
        long n = num;
        int answer = 0;
        
        while (n != 1 && answer < 500) {
            answer++;
            
            if(isOdd(n)) {
                n = n * 3 + 1;
            } else {
                n /= 2;
            }
        }
        
        return (n != 1) ? -1 : answer;
    }
    
    private static boolean isOdd(long num) {
        return num % 2 != 0;
    }
}